db/crud.go

```diff
@@ -567,13 +567,13 @@ func (db *Database) getRevFromDoc(doc *document, revid string, listRevisions boo
 
 // Returns the body of the asked-for revision or the most recent available ancestor.
 // Does NOT fill in _id, _rev, etc.
-func (db *Database) getAvailableRev(doc *document, revid string) (Body, error) {
+func (db *Database) getAvailableRev(doc *document, revid string) (Body, string, error) {
```

Comment:
getRevision is already populating _id, _rev and _attachments in the returned Body, so it shouldn't be necessary to additionally return revid from this method call - it's in the Body.

Reply:
I added this just to be safe. As can be seen I don't think the body is actually supposed to have the rev even though it does given this line:
`// Does NOT fill in _id, _rev, etc.`

Reply:
I think that 'Does NOT fill in...' comment is incorrect, since it's turning around and calling getRevision, which states that it *does* add those properties.  It would be good to clean that up.